### PR TITLE
Re-enable continuous integration

### PR DESCRIPTION
**Pull request by felixjendrusch**
This re-enables continuous integration builds on Travis CI by [explicitly setting build dependencies](e396263) as [described by `xctool`](https://github.com/facebook/xctool#continuous-integration) and [restricting tests to available targets](2bc64e0) (Mac and iOS).

### CODE REVIEW

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme [hunk 1]
```diff
@@ -3,8 +3,8 @@
    LastUpgradeVersion = "0700"
    version = "1.3">
    <BuildAction
-      parallelizeBuildables = "YES"
-      buildImplicitDependencies = "YES">
+      parallelizeBuildables = "NO"
+      buildImplicitDependencies = "NO">
```
Comment (NachoSoto) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Can you explain why this change?

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme [hunk 2]
```diff
@@ -3,8 +3,8 @@
    LastUpgradeVersion = "0700"
    version = "1.3">
    <BuildAction
-      parallelizeBuildables = "YES"
-      buildImplicitDependencies = "YES">
+      parallelizeBuildables = "NO"
```
Comment (NachoSoto) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Can you explain the reason for this change?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+language: objective-c
+osx_image: xcode7.1
+before_install: true
+install: true
+git:
+  submodules: false
+script: script/cibuild ReactiveCocoa-Mac ReactiveCocoa-iOS
```
Comment (NachoSoto) [inline .travis.yml]: Ooooh nice, I didn't realize we could specify these :+1:

**Comment by NachoSoto** [comment]
Thanks for the PR! I'm excited to have this back again :D

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 1 above)
Comment (felixjendrusch) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: As mentioned, `xctool` requires dependencies to be specified explicitly, meaning there should be no implicit dependencies. Disabling the finding of implicit dependencies should also help avoiding any confusion.

More related (possibly outdated) background:
- http://www.openradar.me/20490378#ag9zfm9wZW5yYWRhci1ocmRyFAsSB0NvbW1lbnQYgICA4LKcjwsM

  > I found that Implicit Dependencies would only mess up if I didn't have all dependencies explicitly configured. I I made sure that all dependencies where explicit, the problem would go away.

- http://pewpewthespells.com/blog/managing_xcode.html#scheme-action

  > However there are some edge-cases to this behavior. If you find yourself in a situation where you cannot rely on Xcode's ability to resolve the necessary target dependencies correctly, then both "Parallelize Build" and "Find Implicit Dependencies" should be disabled on the scheme.

- http://andymolloy.net/workspace-interproject-dependencies-and-xcodebuild/

  > If you have an Xcode workspace containing multiple projects, and a scheme in that workspace depends on targets in more than one of those projects, Xcode’s own automatic dependency detection may well work, but xcodebuild will fail. [...] I also had to uncheck the Parallelize Targets box.

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 2 above)
Comment (felixjendrusch) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: See https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2545#discussion_r44449283.

Comment (NachoSoto) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: That explains `buildImplicitDependencies` (which I agree with, I'm :+1: for explicitness), but why `parallelizeBuildables`?

**Comment by NachoSoto** [comment]
Part of me wonders why we're still using `xctool`. It does much more than we need, ideally we just use a tool that works **on top of xcodebuild** (same philosophy as `Carthage`), like [`scan`](https://github.com/fastlane/scan).

File: ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme (hunk 2 above)
Comment (felixjendrusch) [inline ReactiveCocoa.xcodeproj/xcshareddata/xcschemes/ReactiveCocoa-Mac.xcscheme]: Xcode might decide to build `Result` and `ReactiveCocoa` in parallel because there is no target dependency for the former on the latter. This should never happen because `ReactiveCocoa` depends on `Result`. A proper target dependency would require a project reference of the `Result` project on the `ReactiveCocoa` project, which would also add the schemes of the former to the latter.

**Comment by felixjendrusch** [comment]
I agree on `xctool` vs. something that works on top of `xcodebuild` but would prefer to not depend on Ruby, Gems, Fastlane, ... so yeah, something "like scan".

**Comment by NachoSoto** [comment]
We can just invoke `xcodebuild` directly then, it's not that much work.

**Comment by felixjendrusch** [comment]
I like the idea of having reusable build scripts though, like [`jspahrsummers/objc-build-scripts`](https://github.com/jspahrsummers/objc-build-scripts), so maybe the switch from `xctool` to `xcodebuild` should be proposed there?

**Comment by NachoSoto** [comment]
That's a fair point.

I'm going to merge this since it's useful nonetheless! We can consider updating the build script at some other point.

Thanks a lot :star: :green_heart:
